### ISSUE TITLE
A "Scrubbing Delay"

### ISSUE THREAD

**Issue by baconpaul** [surge-synthesizer/surge-rack#632]
https://community.vcvrack.com/t/idea-for-a-module-buffer-with-automatable-playhead/18686/24

have you tried surge xt? We have 2 delays in there - a regular digital delay but also a “tuned delay” which I put together for physical modeling usages. but they all have the same internal line

I think that tuned delay panel is exactly what you want. Stereo in and out; Single CV; and then a couple of range knobs. Or maybe a pair of CV in and pair of range knobs. The trick is, unlike the tuned delay, it would be monophonic and the CV would be interpreted as linear delay not v/oct.

We’ve frozen XT for the first release now but lemme open an issue pointing to this thread and if no-one else comes up with a better idea, maybe we can add a third delay in a future release.

**Comment by MaZe6K** [comment]
Hi! It's me. Here I'm this account. :) I'll try to summarize this type of delay here:

This delay should have a buffer that can be navigated with a CV input by applying a tempo based offset to the read position.
It should have a sync input for tempo detection and scale the CV in a rhythmically useful manner (e.g. 1V/quarter note).

With this you could then not only jump through the buffer, effectively slicing and rearranging it, but also perform all kinds of moves, like slowdowns, scratches, pitch warble etc.

Together with Shapemaster it would be like a full clone of GrossBeat (or MRhythmizer, or TimeShaper (which are clones of GrossBeat as well)), but in a modular environment it could of course also be used with other types of modulators.

**Comment by baconpaul** [comment]
Yeah cool idea. @Torinbob I'm sure would love to see more people doing shapemaster + surge combos also :)

This module is super duper easy to write. We have everything already - clocks, tempo sync with musical intent, delay lines. I've marked it into the "XT 1.1" milestone and will take a shot at it once we have the first version in the library.
